Add primitive string type. (#18973)
coq/coq: doc/sphinx/language/core/primitive.rst

changes dynamically upon each get and set call: the current list of modifications
is applied to the native array and the lists of modifications of the other versions
are updated so that they still represent the same values.

.. _primitive-string:

> Primitive Strings

Author: Done.

----------------

> The language of terms features immutable strings as values. The type of such a
> value is *axiomatized*; it is declared through the following sentence (excerpt
> from the :g:`PrimString` module):

Reviewer: ```suggestion
The language of terms supports immutable strings as values. Primitive strings
are *axiomatized*.  The type is declared through the following sentence (excerpt
from the :g:`PrimString` module):
```

"axiomatized" seems like the wrong term here, certainly not the usual meaning (where are the axioms and what are they?).  Is there a better term?  "built-in"?  Also seemed like it should be 2 sentences instead of joined with a semicolon.

Author: Thanks for the comments! I dunno what terminology is best here: this sentence (like the rest of the section) is basically copy-pasted from the section on primitive floating-point numbers. Maybe we should just keep the different sections on primitives uniform so that the whole page can be edited more easily when someone gets to it? (This documentation page could use some love, given the amount of duplication it contains, but probably not in this PR.)

Reviewer: Yeah, I noticed it was copy-paste and that the wording wasn't too helpful.  But more work than I (or you) want to take on today.  I would have to research the feature and do experiments to know what to say.  Agreed, not in this PR.

.. coqdoc::

>    Primitive string := #string_type.

Reviewer: How about making this a ..coqtop instead, which will check the syntax?

Author: Let's not do that for consistency with the other sections.
